Simplified/Reader2/LCP/LCPAuthenticationViewController.swift

```diff
@@ -0,0 +1,214 @@
+//
+//  LCPAuthenticationViewController.swift
+//  r2-testapp-swift
+//
+//  Created by Mickaël Menu on 01.03.19.
+//
+//  Copyright 2019 Readium Foundation. All rights reserved.
+//  Use of this source code is governed by a BSD-style license which is detailed
+//  in the LICENSE file present in the project repository where this source code is maintained.
+//
+
+#if LCP
+
+import SafariServices
+import UIKit
+import ReadiumLCP
+
+
+protocol LCPAuthenticationDelegate: AnyObject {
+    
+    func authenticate(_ license: LCPAuthenticatedLicense, with passphrase: String)
+    func didCancelAuthentication(of license: LCPAuthenticatedLicense)
+
+}
+
+class LCPAuthenticationViewController: UIViewController {
+    
+    weak var delegate: LCPAuthenticationDelegate?
+    
+    @IBOutlet weak var scrollView: UIScrollView!
+    @IBOutlet weak var hintLabel: UILabel!
+    @IBOutlet weak var promptLabel: UILabel!
+    @IBOutlet weak var messageLabel: UILabel!
+    @IBOutlet weak var passphraseField: UITextField!
+    @IBOutlet weak var supportButton: UIButton!
+    
+    private let license: LCPAuthenticatedLicense
+    private let reason: LCPAuthenticationReason
+    private let supportLinks: [(Link, URL)]
```

Comment:
can we move these to the delegate class maybe? they don't really belong in the VC I think. I think it'd be better to expand the delegate api instead.

Reply:
`LCPAuthenticationReason` is a parameter that defines why we are showing the view (either we can't find the passphrase matching the license or the passphrase is not correct).
`LCPAuthenticatedLicense` is parsed .lcpl data, containing license information for this view.
I think both are the part of this VC, and it is rather hard to remove them from the view controller - it just shows properties from that parameters.

Reply:
Can we just have the delegate derive the supportLinks instead of having the VC do that? We don't even need a property on the VC I think: it seems like you're deriving the support links from the license, which the delegate aka LCP Service already has.

Reply:
I'm just trying to keep the VC "pure", because once you start adding properties and bits of logic here and there, it's a slippery slope to get to Massive View Controllers... then we'll spend 2 weeks refactoring them as I just did for the sign-in VCs.

Reply:
added an adapter class for license data